src/simplewallet/simplewallet.cpp

```diff
@@ -4786,20 +4783,28 @@ bool simple_wallet::try_connect_to_daemon(bool silent, uint32_t* version)
   uint32_t version_ = 0;
   if (!version)
     version = &version_;
-  if (!m_wallet->check_connection(version))
+  bool wallet_is_outdated, daemon_is_outdated = false;
+  if (!m_wallet->check_connection(version, NULL, 200000, &wallet_is_outdated, &daemon_is_outdated))
   {
     if (!silent)
     {
       if (m_wallet->is_offline())
         fail_msg_writer() << tr("wallet failed to connect to daemon, because it is set to offline mode");
+      else if (wallet_is_outdated)
```

Comment:
It seems all errors always take this branch, the elseif/else below won't activate. I tried using a v0.18 wallet on a couple v0.17 nodes as well (including the one linked: http://support.ablative.hosting:18081), same thing.

These all return the old wallet error:
- `monerod` online, but it's v0.17.x.x
- `monerod` online, but connected to wrong port
- `monerod` offline, either local or remote
- random fake ip: `1.2.3.4:18081`
- bad ip: `1:18081` (returns wallet outdated error after hanging for a bit)